src/MeoAssistant/AutoRecruitTask.cpp

```diff
@@ -43,189 +111,365 @@ asst::AutoRecruitTask& asst::AutoRecruitTask::set_skip_robot(bool skip_robot) no
     return *this;
 }
 
+asst::AutoRecruitTask& asst::AutoRecruitTask::set_set_time(bool set_time) noexcept
+{
+    m_set_time = set_time;
+    return *this;
+}
+
 bool asst::AutoRecruitTask::_run()
 {
-    if (!check_recruit_home_page()) {
-        return false;
+    if (is_calc_only_task()) {
+        int tags_selected = 0;
+        bool force_skip = false;
+        return recruit_calc_task(force_skip, tags_selected);
     }
 
-    analyze_start_buttons();
+    if (!recruit_begin()) return false;
 
-    // 不使用加急许可的正常公招
-    for (; m_cur_times < m_max_times && m_cur_times < m_start_buttons.size(); ++m_cur_times) {
-        if (need_exit()) {
-            return false;
-        }
-        if (!recruit_index(m_cur_times)) {
-            return false;
-        }
-    }
-    if (!m_use_expedited) {
-        return true;
+    if (!check_recruit_home_page()) {
+        return false;
     }
-    Log.info("ready to use expedited");
-    // 使用加急许可
-    for (; m_cur_times < m_max_times; ++m_cur_times) {
-        if (need_exit()) {
-            return false;
-        }
-        if (!recruit_now()) {
-            return true;
-        }
-        if (need_exit()) {
-            return false;
-        }
-        analyze_start_buttons();
-        if (!recruit_index(0)) {
-            return false;
+
+    if (!m_use_expedited)
+        analyze_start_buttons(); // analyze once only
+
+    static constexpr size_t slot_retry_limit = 3;
+
+    size_t slot_fail = 0;
+    size_t recruit_times = 0; // how many times has the confirm button been pressed, NOT expedited plan used
+    while ((m_use_expedited || !m_pending_recruit_slot.empty()) && recruit_times != m_max_times) {
```

Comment:
Comparison of integers of different signs: 'size_t' (aka 'unsigned long long') and 'int' [clang-diagnostic-sign-compare]